### PR TITLE
JP2KAK: add support for ARM builds

### PR DESCRIPTION
**Pull request by gillins**
## What does this PR do?

Fixes building against the Kakadu driver for ARM CPUs on Linux and OSX. Tested on Linux aarch64 and x64. 

Unfortunately I don't have access to an Apple M1 to check the OSX changes with.

## What are related issues/pull requests?

None

## Tasklist

 - [ ] ADD YOUR TASKS HERE
 - [ ] Add test case(s)
 - [ ] Add documentation
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

File: cmake/modules/packages/FindKDU.cmake [hunk 1]
```diff
@@ -39,6 +39,24 @@ if(KDU_ROOT AND EXISTS "${KDU_ROOT}/coresys")
         find_library(KDU_AUX_LIBRARY ${KDU_AUX_SHARED_LIB_NAME}
                      PATH ${KDU_ROOT}/lib/Linux-x86-64-gcc)
 
+    elseif("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "aarch64" AND "${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
+        file(READ "${KDU_ROOT}/coresys/make/Makefile-Linux-arm-64-gcc" CORESYS_MAKEFILE_CONTENTS)
```
Comment (rouault) [inline cmake/modules/packages/FindKDU.cmake:43]: as this logic is mostly common with the x86_64 case but the Linux-x86-64-gcc vs Linux-arm-64-gcc difference, couldn't that be factored into a function / macro ?

File: cmake/modules/packages/FindKDU.cmake [hunk 2]
```diff
@@ -50,6 +68,17 @@ if(KDU_ROOT AND EXISTS "${KDU_ROOT}/coresys")
         find_library(KDU_AUX_LIBRARY ${KDU_AUX_SHARED_LIB_NAME}
                      PATH ${KDU_ROOT}/lib/Mac-x86-64-gcc)
 
+    elseif("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "arm64" AND "${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
+        file(READ "${KDU_ROOT}/coresys/make/Makefile-Mac-arm-64-gcc" CORESYS_MAKEFILE_CONTENTS)
```
Comment (rouault) [inline cmake/modules/packages/FindKDU.cmake:72]: same remark

File: frmts/jp2kak/CMakeLists.txt [hunk 3]
```diff
@@ -2,4 +2,13 @@ add_gdal_driver(TARGET gdal_JP2KAK SOURCES jp2kakdataset.cpp jp2kak_headers.h jp
...
 gdal_target_link_libraries(gdal_JP2KAK PRIVATE KDU::KDU)
-target_compile_definitions(gdal_JP2KAK PRIVATE -DKDU_X86_INTRINSICS)
+
+# this appears necessary on x64 machines to match the compilation of the library
+# But for ARM we don't need to define KDU_NEON_INTRINSICS, because 
+# according to Compiling_Instructions.txt: "Note that this macro does not normally need to
+#             be set explicitly, since it is automatically configured
+#             by "kdu_arch.h" if the usual "__ARM_NEON__" is found"
+# __ARM_NEON__ appears to be defined when NEON extensions available (https://developer.arm.com/documentation/dui0491/i/Compiler-specific-Features/Predefined-macros)
+if("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "x86_64")
```
Comment (rouault) [inline frmts/jp2kak/CMakeLists.txt:12]: For Windows x86_64 builds:
```suggestion
if("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "x86_64" OR "${CMAKE_SYSTEM_PROCESSOR}" MATCHES "AMD64")
```

**Comment by gillins** [comment]
Thanks @rouault I've done as you suggested and as a bonus added MinGW support. Let me know if there is anything else.

**Comment by rouault** [comment]
thanks!

**Comment by rouault** [comment]
The backport to `release/3.5` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
error: could not apply a3d4734f2f... put KDU lib searching logic into a macro
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".

```
</details>

<details>
<summary>stdout</summary>

```
Auto-merging cmake/modules/packages/FindKDU.cmake
[backport-6222-to-release/3.5 4f34b63dd9] add support for ARM Makefiles etc for building JP2KAK
 Author: Sam Gillingham <gillingham.sam@gmail.com>
 Date: Thu Aug 18 10:29:34 2022 +1000
 2 files changed, 38 insertions(+), 1 deletion(-)
[backport-6222-to-release/3.5 74e689bf45] add comment about __ARM_NEON__
 Author: Sam Gillingham <gillingham.sam@gmail.com>
 Date: Thu Aug 18 12:47:34 2022 +1000
 1 file changed, 1 insertion(+)
Auto-merging cmake/modules/packages/FindKDU.cmake
CONFLICT (content): Merge conflict in cmake/modules/packages/FindKDU.cmake

```
</details>

To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-release/3.5 release/3.5
# Navigate to the new working tree
cd .worktrees/backport-release/3.5
# Create a new branch
git switch --create backport-6222-to-release/3.5
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick e6ec49bbed42bb3307595d0b1beaffd444571026,2719fb4c3b7759c06aa0650e765cead646825de7,a3d4734f2f80563405e5e770b595fd0817435d5a
# Push it to GitHub
git push --set-upstream origin backport-6222-to-release/3.5
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-release/3.5
```
Then, create a pull request where the `base` branch is `release/3.5` and the `compare`/`head` branch is `backport-6222-to-release/3.5`.

**Pull request merged**